### PR TITLE
Cancel chartconfig resource if cluster-operator cert or chartconfig CRD do not exist

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

Cancel reconciliation of chartconfig resource if the cluster-operator API cert is not found. Or if the chartconfig CRD is not found in the guest cluster.

In both cases we cancel as we can't continue and will retry in the next execution. For the chartconfig CRD there will be a delay until chart-operator is running in the guest cluster since it creates the CRD.

### CODE REVIEW

File: pkg/v1/guestcluster/guestcluster.go [hunk 1]
```diff
@@ -83,7 +83,9 @@ func (s *Service) newKubernetesRestConfig(ctx context.Context, clusterID, apiDom
 	s.logger.LogCtx(ctx, "level", "debug", "message", "looking for certificate to connect to the guest cluster")
 
 	operatorCerts, err := s.certsSearcher.SearchClusterOperator(clusterID)
-	if err != nil {
+	if certs.IsTimeout(err) {
```
Comment (xh3b4sd) [inline pkg/v1/guestcluster/guestcluster.go]: Which timeout is configured? The default is 90 seconds IIRC but this is too long here IMO. 3 should be good to keep the reconciliation quick. WDYT?

Comment (rossf7) [reply pkg/v1/guestcluster/guestcluster.go]: Agree it should be quick. I used 5 secs when testing but 3 secs is better.

Problem is the resource isn't configured yet. I don't want to do that until the chart resource that installs chart-operator is also ready.

Comment (xh3b4sd) [reply pkg/v1/guestcluster/guestcluster.go]: Ok cool.

Comment (rossf7) [reply pkg/v1/guestcluster/guestcluster.go]: I'll add the configuration but hold off enabling the resource until the chart resource is ready.

Comment (rossf7) [reply pkg/v1/guestcluster/guestcluster.go]: Scratch that I think it's better to do this when the resource is enabled. Going with this.

**Pull request merged**